keps/sig-storage/1472-storage-capacity-tracking/README.md

```diff
@@ -585,34 +577,82 @@ instead. Obsolete objects need to be removed.
 
 To ensure that `CSIStorageCapacity` objects get removed when the driver
 deployment gets removed before it has a chance to clean up, each
-`CSIStorageCapacity` object needs an [owner
+`CSIStorageCapacity` object should have an [owner
 reference](https://godoc.org/k8s.io/apimachinery/pkg/apis/meta/v1#OwnerReference).
 
-For central provisioning, that has to be the Deployment or StatefulSet
-that defines the provisioner pods. That way, provisioning can
-continue seamlessly when there are multiple instances with leadership
-election.
-
-For deployment with a daemon set, making the individual pod the owner
-is better than the daemon set as owner because then other instances do
-not need to figure out when to remove `CSIStorageCapacity` objects
-created on another node. It is also better than the node as owner
-because a driver might no longer be needed on a node although the node
-continues to exist (for example, when deploying a driver to nodes is
-controlled via node labels).
-
-While external-provisioner runs, it needs to update and potentially
+The owner should be the higher-level app object which defines the provisioner pods:
+- the Deployment or StatefulSet for central provisioning
+- the DaemonSet for distributed provisioning
+
+This way, provisioning and pod scheduling can continue seamlessly when
+there are multiple instances with leadership election or the driver
+gets upgraded.
+
+This ownership reference is optional. It gets enabled with
```

Comment:
What is the reason for making it optional?

Reply:
It can't be mandatory because drivers that aren't deployed inside Kubernetes have nothing that can serve as owner.

We could remove it complete, but I find it useful because admins don't need to worry about manually removing objects after removing the "normal" driver objects that they know about (StatefulSet, DaemonSet, etc.).

That leaves keeping it as an optional feature.

Reply:
Would be useful to provide this context in the KEP itself.

Reply:
Added.